### PR TITLE
Added option page to select linkStyle

### LINKED ISSUE #12: Configure link from reference to inline
Hello!
I am not an expert when it comes to Chrome Extension but I would like to be able to configure how I want my markdown to look like. For example, I prefer to get an inline markdown for the link vs the reference one this extension uses.

is it possible to add a configuration for that?
Thanks

### PR DESCRIPTION
**Pull request by gianarb**
I would like to change the linkStyle. Previously it was fixed to
referenced but very often I want the inline version of it.

This PR drafts an options page that contains a select box to pick the one
you prefer an end-user.

Fixed #12

Thanks a lot for this plugin, easy and cool!

### CODE REVIEW

File: src/options.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <title>Grammarly to Markdown: Options</title>
+        <link rel='stylesheet' href='./styles/main.css' />
+    </head>
+    <body>
+        <h1>Grammarly to Markdown</h1>
+
+        <label for="turndown-linkSyle">Choose a car:</label>
```
Comment (brunoluiz) [inline src/options.html:10]:
```suggestion
        <label for="turndown-linkSyle">Choose link style:</label>
```

File: src/js/options.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Saves options to chrome.storage
+function save_options() {
```
Comment (brunoluiz) [inline src/js/options.js:2]: Just to be consistent with the rest of the code (probably will need adjustments where it is referenced)

```suggestion
const saveOptions = () => {
```

File: src/js/options.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  var linkStyle = document.getElementById('turndown-linkSyle').value;
+  chrome.storage.sync.set({
+    linkStyle: linkStyle
+  });
+}
+
+function restore_options() {
```
Comment (brunoluiz) [inline src/js/options.js:9]: Just to be consistent with the rest of the code (probably will need adjustments where it is referenced)
```suggestion
const restoreOptions = () => {
```

File: src/js/options.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  // Use default value color = 'red' and likesColor = true.
```
Comment (brunoluiz) [inline src/js/options.js:10]:
```suggestion
```

File: src/js/options.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  chrome.storage.sync.get({
+    linkStyle: "referenced"
+  }, function(items) {
```
Comment (brunoluiz) [inline src/js/options.js:13]:
```suggestion
  }, (items) => {
```

File: src/js/popup.js [hunk 6]
```diff
@@ -43,26 +26,48 @@ const setMessage = (data) =>
...
-          type: MESSAGE_ERROR,
+
+    chrome.storage.sync.get({
+      headingStyle: "atx",
+      linkStyle: "referenced",
+      bulletListMarker: "-",
+      br: "",
+    }, function(items){
+        console.log(items)
+        var turndownService = new TurndownService(items);
```
Comment (brunoluiz) [inline src/js/popup.js:37]:
```suggestion
    }, (items) => {
        const turndownService = new TurndownService(items);
```

File: src/js/popup.js [hunk 7]
```diff
@@ -43,26 +26,48 @@ const setMessage = (data) =>
 document.addEventListener("DOMContentLoaded", function () {
   chrome.tabs.query({ active: true, currentWindow: true }, function (tabs) {
     chrome.tabs.sendMessage(tabs[0].id, {}, function (response = {}) {
-      const { content, title } = response;
-      if (!content) {
-        return setMessage({
-          msg: "🚨 Error on getting content",
-          type: MESSAGE_ERROR,
+
```
Comment (brunoluiz) [inline src/js/popup.js:29]:
```suggestion
```

File: src/js/popup.js [hunk 8]
```diff
@@ -43,26 +26,48 @@ const setMessage = (data) =>
...
-        return setMessage({
-          msg: "🚨 Error on markdown parsing",
-          type: MESSAGE_ERROR,
+        turndownService.addRule("h2", {
+          filter: ["h2"],
+          replacement: (content) => `### ${content}\n`,
         });
-      }
+
+
```
Comment (brunoluiz) [inline src/js/popup.js:48]: Usually I try to avoid having too much spacing between sections.

```suggestion
```

File: src/js/options.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    document.getElementById('turndown-linkSyle').value = items.linkStyle;
+  });
+}
+document.addEventListener('DOMContentLoaded', restore_options);
+document.getElementById('save').addEventListener('click', save_options);
```
Comment (brunoluiz) [inline src/js/options.js:18]:
```suggestion
document.getElementById('save').addEventListener('click', saveOptions);
```

File: src/js/options.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+function restore_options() {
+  // Use default value color = 'red' and likesColor = true.
+  chrome.storage.sync.get({
+    linkStyle: "referenced"
+  }, function(items) {
+    document.getElementById('turndown-linkSyle').value = items.linkStyle;
+  });
+}
+document.addEventListener('DOMContentLoaded', restore_options);
```
Comment (brunoluiz) [inline src/js/options.js:17]:
```suggestion
document.addEventListener('DOMContentLoaded', restoreOptions);
```

**Review by brunoluiz — CHANGES_REQUESTED** [review]
Thanks for contributing! I was thinking about adding a configuration pages last week -- You read my mind 😅

Amazing job overall 🙌 I left some suggested changes on the PR and probably I will get some time to test it later this week.

File: src/js/options.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
+// Saves options to chrome.storage
+function save_options() {
+  var linkStyle = document.getElementById('turndown-linkSyle').value;
```
Comment (brunoluiz) [inline src/js/options.js:3]:
```suggestion
  const linkStyle = document.getElementById('turndown-linkSyle').value;
```

**Comment by brunoluiz** [comment]
By the way, I am not sure why CircleCI didn't run tests/linter on your pull request. Mind running it manually? `npm run lint` should do 😉

**Comment by gianarb** [comment]
Updated. Thanks for your review @brunoluiz ! I am not fluent in Javascript as you can imagine, but the `lint` is green now and I am using this version of the extension currently, it works well.

**Review by brunoluiz — APPROVED** [review]

**Comment by brunoluiz** [comment]
Will prepare a release soon 😉

**Pull request merged**